### PR TITLE
[bitnami/fluentd] feat: allow addition of servicemonitor labels and annotations

### PR DESCRIPTION
**Pull request by aslafy-z**
**Description of the change**

Adds values to add labels and annotations on ServiceMonitor.

**Benefits**

Allow users to add labels and annotations on ServiceMonitor.

**Possible drawbacks**

**Applicable issues**

  - fixes #

**Additional information**

**Checklist** 
- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[bitnami/chart]`)

:warning: Keep in mind that if you want to make changes to the kubeapps chart, please implement them in the [kubeapps repository](https://github.com/kubeapps/kubeapps/tree/master/chart/kubeapps). This is only a synchronized mirror.

### CODE REVIEW

**Review by carrodher — COMMENTED** [review]
Thanks for your contribution, I have left a couple of comments/suggestions

File: bitnami/fluentd/README.md [hunk 1]
```diff
@@ -212,6 +212,8 @@ The following tables lists the configurable parameters of the fluentd chart and
...
 | `metrics.serviceMonitor.scrapeTimeout`          | Timeout after which the scrape is ended                                                                        | `nil` (Prometheus Operator default value)                                                               |
 | `metrics.serviceMonitor.selector`               | Prometheus instance selector labels                                                                            | `nil`                                                                                                   |
+| `metrics.serviceMonitor.labels`                 | ServiceMonitor extra labels                                                                                    | `nil`                                                                                                   |
+| `metrics.serviceMonitor.annotations`            | ServiceMonitor annotations                                                                                     | `nil`                                                                                                   |
```
Comment (aslafy-z) [inline bitnami/fluentd/README.md:216]:
```suggestion
```

File: bitnami/fluentd/templates/servicemonitor.yaml [hunk 2]
```diff
@@ -10,6 +10,17 @@ metadata:
     {{- range $key, $value := .Values.metrics.serviceMonitor.selector }}
     {{ $key }}: {{ $value | quote }}
     {{- end }}
+  {{- if .Values.metrics.serviceMonitor.labels }}
+    {{- range $key, $value := .Values.metrics.serviceMonitor.labels }}
+    {{ $key }}: {{ $value | quote }}
+    {{- end }}
+  {{- end }}
```
Comment (aslafy-z) [inline bitnami/fluentd/templates/servicemonitor.yaml:17]:
```suggestion
{{- if .Values.metrics.serviceMonitor.labels }}
{{- toYaml .Values.metrics.serviceMonitor.labels | nindent 4 }}
{{- end }}
```

File: bitnami/fluentd/templates/servicemonitor.yaml [hunk 3]
```diff
@@ -10,6 +10,17 @@ metadata:
...
+  {{- if .Values.metrics.serviceMonitor.annotations }}
+  annotations:
+    {{- range $key, $value := .Values.metrics.serviceMonitor.annotations }}
+    {{ $key }}: {{ $value | quote }}
+    {{- end }}
+  {{- end }}
```
Comment (aslafy-z) [inline bitnami/fluentd/templates/servicemonitor.yaml:23]:
```suggestion
```

Comment (aslafy-z) [inline bitnami/fluentd/templates/servicemonitor.yaml:23]:
```suggestion
```

File: bitnami/fluentd/templates/servicemonitor.yaml [hunk 4]
```diff
@@ -10,6 +10,17 @@ metadata:
...
+  {{- if .Values.metrics.serviceMonitor.labels }}
+    {{- range $key, $value := .Values.metrics.serviceMonitor.labels }}
+    {{ $key }}: {{ $value | quote }}
+    {{- end }}
+  {{- end }}
+  {{- if .Values.metrics.serviceMonitor.annotations }}
+  annotations:
+    {{- range $key, $value := .Values.metrics.serviceMonitor.annotations }}
+    {{ $key }}: {{ $value | quote }}
+    {{- end }}
```
Comment (aslafy-z) [inline bitnami/fluentd/templates/servicemonitor.yaml:22]:
```suggestion
{{- toYaml .Values.metrics.serviceMonitor.annotations | nindent 4 }}
```

File: bitnami/fluentd/templates/servicemonitor.yaml [hunk 5]
```diff
@@ -10,6 +10,13 @@ metadata:
     {{- range $key, $value := .Values.metrics.serviceMonitor.selector }}
     {{ $key }}: {{ $value | quote }}
     {{- end }}
+{{- if .Values.metrics.serviceMonitor.labels }}
+{{- toYaml .Values.metrics.serviceMonitor.labels | nindent 4 }}
+{{- end }}
+  {{- if .Values.metrics.serviceMonitor.annotations }}
+  annotations:
+{{- toYaml .Values.metrics.serviceMonitor.annotations | nindent 4 }}
+  {{- end }}
```
Comment (marcosbc) [inline bitnami/fluentd/templates/servicemonitor.yaml:19]: These lines seem to be wrongly indented, could you re-check them?
```suggestion
    {{- if .Values.metrics.serviceMonitor.labels }}
    {{- toYaml .Values.metrics.serviceMonitor.labels | nindent 4 }}
    {{- end }}
  {{- if .Values.metrics.serviceMonitor.annotations }}
  annotations: {{- toYaml .Values.metrics.serviceMonitor.annotations | nindent 4 }}
  {{- end }}
```

Comment (aslafy-z) [reply bitnami/fluentd/templates/servicemonitor.yaml:19]: Well, it looks good to me. The linter is OK

Comment (marcosbc) [reply bitnami/fluentd/templates/servicemonitor.yaml:19]: The problem is that they are not indented consistenly (i.e. check the rest of the file). Could you change it? Thanks!

Comment (aslafy-z) [reply bitnami/fluentd/templates/servicemonitor.yaml]: Sorry, I was missing the definition of `nindent` over `indent`.
Thanks for the head-up!
It's fixed now.

**Review by marcosbc — APPROVED** [review]
LGTM! Thanks for making the changes!

**Pull request merged**